### DISCUSSION TITLE
Var, Call and other constructs in default rule

### DISCUSSION THREAD

**Discussion by anakrish** [open-policy-agent/community#338] [OPA and Rego] [answered]
I'm trying to understand which constructs are supported in a `default` rule.

The following errors out ` rego_parse_error: illegal default rule (value cannot contain var)` https://play.openpolicyagent.org/p/t21XFFhcNN
```
package test
a = 8
default b = [a]
```

However, constructs work if enclosed within a comprehension in the default rule's value:
https://play.openpolicyagent.org/p/lOdShubvHT
```
package test
a = 8
default b = [x | startswith("pq", "p"); x = a ]
```

Is there a design rationale as to why the constructs are supported when enclosed within a comprehension?

**Comment by anderseknert** [comment, marked as answer]
Hi @anakrish 👋 That's an interesting observation! I would believe that to be a bug. Trying your example I had no problems depending on `input`, `data` and built-in functions inside of the comprehension, creating non-deterministic default values. I don't think that should ever work. I'll create an issue.

**Reply by anderseknert** [reply]
Created issue https://github.com/open-policy-agent/opa/issues/5645
